synapse/storage/databases/main/devices.py

```diff
@@ -854,7 +868,9 @@ async def get_all_device_list_changes_for_remotes(
         if last_id == current_id:
             return [], current_id, False
 
-        def _get_all_device_list_changes_for_remotes(txn):
+        def _get_all_device_list_changes_for_remotes(
+            txn: Cursor,
```

Comment:
Ist there any reason why this is a `Cursor` and not a `LoggingTransaction`?

Reply:
Either is fine. This function will eventually be called with a `LoggingTransaction`, but it only needs a `Cursor` to work. (Though in some places we do define transaction functions that use `LoggingTransaction` methods I think?)

I don't fully understand the machinery behind our `LoggingTransaction` and `LoggingDatabaseConnection` and all of this though... happy to be educated!!

Reply:
I don't have an explanation and so I asked for one.

Reply:
Using `Cursor` is just more general, if you're passing into any of the `simple_...` methods then you generally need to use `LoggingTransaction` IIRC.